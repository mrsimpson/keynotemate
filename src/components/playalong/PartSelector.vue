<template>
  <div class="part-selector">
    <label for="part-select">Select Part:</label>
    <select
      id="part-select"
      :value="modelValue"
      class="part-select"
      @change="
        $emit('update:modelValue', ($event.target as HTMLSelectElement).value)
      "
    >
      <option v-for="part in parts" :key="part.id" :value="part.id">
        {{ part.name }}
      </option>
    </select>
  </div>
</template>

<script setup lang="ts">
import type { PartInfo } from '../../types/musicxml';

defineProps<{
  modelValue: string;
  parts: PartInfo[];
}>();

defineEmits<{
  (e: 'update:modelValue', value: string): void;
}>();
</script>

<style scoped>
.part-selector {
  margin-bottom: var(--spacing-lg);
}

.part-select {
  margin-left: var(--spacing-sm);
  padding: var(--spacing-sm);
  border: 1px solid #ddd;
  border-radius: var(--radius-sm);
  font-size: 1rem;
}
</style>
