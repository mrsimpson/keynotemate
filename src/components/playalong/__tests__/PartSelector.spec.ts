import { mount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'
import PartSelector from '../PartSelector.vue'
import type { PartInfo } from '@/types/musicxml.ts'

describe('PartSelector Component', () => {
  const mockParts: PartInfo[] = [
    { id: 'part1', name: 'Part 1' },
    { id: 'part2', name: 'Part 2' },
  ]

  it('renders part selector snapshot', () => {
    const wrapper = mount(PartSelector, {
      props: {
        modelValue: 'part1',
        parts: mockParts,
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders part selector with different selected part snapshot', () => {
    const wrapper = mount(PartSelector, {
      props: {
        modelValue: 'part2',
        parts: mockParts,
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
