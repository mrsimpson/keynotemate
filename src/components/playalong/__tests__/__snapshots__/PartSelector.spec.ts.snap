// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PartSelector Component > renders part selector snapshot 1`] = `
"<div data-v-a7238697="" class="part-selector"><label data-v-a7238697="" for="part-select">Select Part:</label><select data-v-a7238697="" id="part-select" class="part-select" value="part1">
    <option data-v-a7238697="" value="part1">Part 1</option>
    <option data-v-a7238697="" value="part2">Part 2</option>
  </select></div>"
`;

exports[`PartSelector Component > renders part selector with different selected part snapshot 1`] = `
"<div data-v-a7238697="" class="part-selector"><label data-v-a7238697="" for="part-select">Select Part:</label><select data-v-a7238697="" id="part-select" class="part-select" value="part2">
    <option data-v-a7238697="" value="part1">Part 1</option>
    <option data-v-a7238697="" value="part2">Part 2</option>
  </select></div>"
`;
